### ISSUE TITLE
Example of "diagnostic"?

### ISSUE THREAD

**Issue by kamahen** [kythe/kythe#3734]
There's no example of `diagnostic` in the schema, and no test case AFAICT.

I infer the following (from looking at `doc` and `documents`) ... is it correct?
```
//- @v defines/binding V
//- V.node/kind variable
//- @v tagged VDiagnostic
//- VDiagnostic.node/kind diagnostic
//- VDiagnostic.message "Unused variable"
int v;
```

In this situation, it seems more natural to instead have `V tagged VDoc`, but the schema says that `tagged` takes `file` or `anchor`.

PS: in the example in https://kythe.io/docs/schema/#documents is `int v;` extraneous?

**Comment by shahms** [comment]
The `int v;` in the `documents` example does seem extraneous.  Your example seems correct, yes.

`{file, anchor} -- tagged --> { diagnostic }`

"documents" doesn't enter into it.  That node is about documenting a particular entity, `diagnostic` is about calling out (generally) automated errors or warnings during analysis.

**Comment by kamahen** [comment]
Is it allowed for something other than a `file` or `anchor` to be `tagged` with a `diagnostic`?

Explanations such as `{file,anchor} — tagged → {diagnostic}` would be helpful elsewhere in the schema doc (e.g. `{doc,anchor} — documents → {...}`).

BTW, I mentioned `documents` because it seemed likely that `diagnostic` would use a similar node/edge structure (`s/documents/tagged/`).

**Comment by creachadair** [comment]
> Is it allowed for something other than a `file` or `anchor` to be `tagged` with a `diagnostic`?

We intentionally didn't restrict it that way—in principle a diagnostic could be attached to anything for which it's relevant. Files and anchors are the most obvious, though, since files and file-ranges are typically what you get from a compiler, and I don't think we have any other specific types that use them right now.

**Comment by kamahen** [comment]
The documentation is misleading (to me anyway) ... https://kythe.io/docs/schema/#tagged says 
Points from
    [anchor], [file]
Points toward
    [diagnostic]

**Comment by creachadair** [comment]
I'd say the documentation reflects existing practice—are there other kinds you think should be included?

**Comment by kamahen** [comment]
I could envision a `variable` being marked with a diagnostic such as `not initialized`, which doesn't have an obvious `anchor`.

But my point is that the documentation seems to say that *only* `anchor` and `file` can be `tagged` with a `diagnostic` whereas @creachadair seems to be saying that these are merely the most common situations. (By comparison, the documentation for `documents` seems clear that it can be for any node.)

BTW, the example for `doc` seems to be missing a line or two:
```
//- @f defines/binding FnF  // missing
//- FnF.node/kind function  // possibly missing
int f(int x, int y) {
```

**Comment by creachadair** [comment]
> But my point is that the documentation seems to say that _only_ `anchor` and `file` can be `tagged` with a `diagnostic` whereas @creachadair seems to be saying that these are merely the most common situations.

That's my recollection of the design discussion that led to the current wording in the schema, anyway I would not object to extending the supported set if there's a use for it. Both `tagged` and `depends` list the kinds we thought were relevant at the time, rather than saying "anything goes" by default, since diagnostics are user-facing and thus have UI implications.

No reason we can't improve the examples and/or add clarifying text, though.

**Comment by shahms** [comment]
fixed via #3962
